### PR TITLE
Fix bintray/upload and add tests.

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**

### CODE REVIEW

File: bin/bintray/upload
Comment (joseph-wakeling-sociomantic) [inline bin/bintray/upload:151]: .... how many times did you re-read this line just in case? :-P

File: test/bintray/upload/1
Comment (joseph-wakeling-sociomantic) [inline test/bintray/upload/1]: Is the space after `--released` intentional here?

File: test/bintray/test
Comment (joseph-wakeling-sociomantic) [inline test/bintray/test:6]: Can you explain these lines?  I'm not sure I follow the intention.

File: bin/bintray/upload
Comment (leandro-lucarella-sociomantic) [reply bin/bintray/upload:151]: Hopefully only once now that we have tests :wink:

File: test/bintray/upload/1
Comment (leandro-lucarella-sociomantic) [reply test/bintray/upload/1]: Not sure, I mean, `--released` should be a date, and it's empty. Not sure if it should be empty, I will check.

File: test/bintray/test
Comment (leandro-lucarella-sociomantic) [reply test/bintray/test:6]: I'm just creating temporary files to be used by the tests...

File: test/bintray/upload/1
Comment (leandro-lucarella-sociomantic) [reply test/bintray/upload/1]: Yeah, this is OKish, but not very clean. The released date is taken from the Git tag date, but in the fabricated tests there is no Git tag, hence it is empty. Not ideal but fixing that could be challenging (we basically should have some face git repo to get the info from).

Comment (leandro-lucarella-sociomantic) [reply test/bintray/upload/1]: Same goes for `--desc=`.

**Comment by leandro-lucarella-sociomantic** [comment]
OK, green.

**Comment by leandro-lucarella-sociomantic** [comment]
I'll merge after I get any approvals and then I release `rc.2`.

File: bin/bintray/upload
Comment (joseph-wakeling-sociomantic) [inline bin/bintray/upload:129]: Just a minor note: this line appears different from what is `echo`'d above.  Is that intentional?

Comment (joseph-wakeling-sociomantic) [reply bin/bintray/upload:129]: (I recognize the line is unchanged from the previous version, but now that we're actually using an official dry-run mode, having the command accurate might be more important.)

**Review by joseph-wakeling-sociomantic — APPROVED** [review]
LGTM.  Let's get that RC on the road ;-)

**Comment by leandro-lucarella-sociomantic** [comment]
Out!

**Comment by joseph-wakeling-sociomantic** [comment]
This now works -- see MXNet package builds:

* rc: https://travis-ci.org/sociomantic-tsunami/mxnet/builds/390741333

* stable: https://travis-ci.org/sociomantic-tsunami/mxnet/builds/390755097

... so I think you can proceed to final release.

**Pull request merged**
